### ISSUE TITLE
add_user_to_group 404 error

### ISSUE THREAD

**Issue by ghoroupi** [pycontribs/jira#975]
**Describe the bug**
The add_user_to_group method is failing with 404 exception
"Specified user does not exist or you do not have required permissions"

All other methods work.  It's just add_user_to_group

**To Reproduce**
```
options = { 'server': self.site_url }
connection = JIRA(options, basic_auth=(self.username, self.token))
for user in user_list:
     connection.add_user_to_group(username=user, group=group)
```
I tried a few options for user:  username@server, username, user full name

**Expected behavior**
Was expecting that one of the options for user would work

If it's an authorization issue, why would this be any different than other methods?

**Stack Trace**
```
  File "/Users/xxxx/IdeaProjects/rsa_dashboard/VENV37/lib/python3.7/site-packages/jira/client.py", line 2994, in add_user_to_group
    r = json_loads(self._session.post(url, params=x, data=payload))
  File "/Users/xxxx/IdeaProjects/rsa_dashboard/VENV37/lib/python3.7/site-packages/jira/resilientsession.py", line 154, in post
    return self.__verb('POST', url, **kwargs)
  File "/Users/xxxx/IdeaProjects/rsa_dashboard/VENV37/lib/python3.7/site-packages/jira/resilientsession.py", line 147, in __verb
    raise_on_error(response, verb=verb, **kwargs)
  File "/Users/xxxx/IdeaProjects/rsa_dashboard/VENV37/lib/python3.7/site-packages/jira/resilientsession.py", line 57, in raise_on_error
    r.status_code, error, r.url, request=request, response=r, **kwargs)
jira.exceptions.JIRAError: JiraError HTTP 404 url: https://XXXSERVERXXX/rest/api/latest/group/user?groupname=XXXGROUPNAMEXXXX
	text: Specified user does not exist or you do not have required permissions
	
```

**Version Information**
Python Interpreter: 3.7.6
jira-python: 2.0.0
OS: MACOS

**Comment by bmarroquin** [comment]
The[ Jira docs](https://docs.atlassian.com/software/jira/docs/api/REST/8.5.9/#api/2/group-addUserToGroup) state the a 404 means either the Users or Group do not exist. If it was an access issue you would get a different 40X error code. Make sure the group the exists as well.

To make sure you use the right username format, you can use connection.search_users(...) and use the name attribute in the response. Searching with . returns all users.

**Comment by studioj** [comment]
assume fixed in the next public release, please reopen if not fixed in >3.0.1
